niftypet/nipet/prj/src/recon.cu

```diff
@@ -71,24 +267,24 @@ void d_eldiv(float * d_inA,
 
 //- - - - - - - - - - - - - - - - - - - - - - - - - - - - -
 
-__global__ void sneldiv(unsigned short *inA,
-	float *inB,
+__global__ void sneldiv(float *inA,
+	unsigned short *inB,
 	int   *sub,
 	int Nprj,
 	int snno)
 {
 	int idz = threadIdx.x + blockDim.x*blockIdx.x;
 	if (!(blockIdx.y<Nprj && idz<snno)) return;
-	// inB > only active bins of the subset
-	// inA > all sinogram bins
-	float a = (float)inA[snno*sub[blockIdx.y] + idz];
-	if (inB[snno*blockIdx.y + idz] == 0) a = 0;
-	else a /= inB[snno*blockIdx.y + idz];//sub[blockIdx.y]
-	inB[snno*blockIdx.y + idz] = a; //sub[blockIdx.y]
+	// inA > only active bins of the subset
+	// inB > all sinogram bins
+	float b = (float)inB[snno*sub[blockIdx.y] + idz];
+	if (FLOAT_WITHIN_EPS(inA[snno*blockIdx.y + idz])) b = 0;
+	else b /= inA[snno*blockIdx.y + idz];//sub[blockIdx.y]
+	inA[snno*blockIdx.y + idz] = b; //sub[blockIdx.y]
 }
 
-void d_sneldiv(unsigned short * d_inA,
-	float * d_inB,
+void d_sneldiv(float *d_inA,
+	unsigned short *d_inB,
```

Comment:
reordering arguments since `dst, src` is the convention everywhere else

Reply:
OK!